pkg/clusterapi/system.go

```diff
@@ -231,6 +234,28 @@ func (c *system) Run(ctx context.Context) error { //nolint:gocyclo
 			}
 		}
 
+		// ASO expects the contents of the cert--not the path--in the env var.
+		// Since .pfx is a binary format, we need to parse it and convert to PEM format.
+		var certPEM string
+		if session.AuthType == azic.ClientCertificateAuth {
+			certPath := session.Credentials.ClientCertificatePath
+			certData, err := os.ReadFile(certPath)
+			if err != nil {
+				return fmt.Errorf("unable to read client certificate contents from %s: %w", certPath, err)
```

Comment:
```suggestion
				return fmt.Errorf("failed to read client certificate contents from %s: %w", certPath, err)
```

nit: for error message consistency